client/state/plugins/installed/test/selectors-ts.ts

```diff
@@ -213,3 +217,51 @@ describe( 'getAllPluginsIndexedBySiteId', () => {
 		expect( pluginsOne ).toBe( pluginsTwo );
 	} );
 } );
+
+describe( 'getFilteredAndSortedPlugins', () => {
+	test( 'Should get an empty array if the requested site is not in the current state', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ nonExistingSiteId ], undefined );
+		expect( plugins ).toHaveLength( 0 );
+	} );
+
+	test( 'Should get an empty array if the plugins for this site are still being requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteThreeId ], undefined );
+		expect( plugins ).toHaveLength( 0 );
+	} );
+
+	test( 'Should get a plugin list of length 3 if both sites are requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], undefined );
+		expect( plugins ).toHaveLength( 3 );
+	} );
+
+	test( 'Should get a plugin list containing jetpack if both sites are requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], undefined );
+		const siteWithPlugin = {
+			[ siteTwoId ]: {
+				active: jetpack.active,
+				autoupdate: jetpack.autoupdate,
+				update: jetpack.update,
+				version: jetpack.version,
+			},
+		};
+		const { active, autoupdate, update, version, ...pluginProperties } = jetpack;
+		expect( plugins ).toEqual(
+			expect.arrayContaining( [ { ...pluginProperties, sites: siteWithPlugin } ] )
+		);
+	} );
+
+	test( 'Should get a plugin list of length 2 if only site 1 is requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId ], undefined );
+		expect( plugins ).toHaveLength( 2 );
```

Comment:
Noting that for filtering, these kinds of tests that only test the length are not necessarily precise. A filter could return the same number of items, but the wrong items, and this test would still pass.

Reply:
I agree, I think these tests could use a PR to clean up and expand them. But for this PR I want to keep them mostly as they were before.

Reply:
Sounds good, I'm fine with following up on this in another PR 👍